src/main/java/gov/cabinetoffice/gap/applybackend/service/GrantAdvertService.java

```diff
@@ -84,6 +87,35 @@ public boolean advertExistsInContentful(final String advertSlug) {
         return advertExists;
     }
 
+    private String getGrantWebpageUrl(final CDAArray contentfulEntry){
+        CDAResource entry = contentfulEntry.items().get(0);
```

Comment:
`entry` here is in fact a `CDAEntry`, not a `CDAResource`. Despite what the library tells us should be returned. This has caused extra (ugly) typecasting below. Not sure how to get around this

<img width="681" alt="Screenshot 2023-12-15 at 15 56 45" src="https://github.com/cabinetoffice/gap-find-applicant-backend/assets/135222889/561be08c-29aa-4164-9d4a-40d9e33f7a03">

Reply:
may as well cast this on the line it's defined

```java
        CDAEntry entry = (CDAEntry) contentfulEntry.items().get(0);
        Map<String, Object> rawFields = entry.rawFields();
```